src/shared/bpf-dlopen.h

```diff
@@ -6,14 +6,34 @@
 #include <bpf/bpf.h>
 #include <bpf/libbpf.h>
 
+/* define required struct to allow newer API with older bpf.h
+ * remove together with pre-1.0 compat helpers below */
+#ifndef bpf_map_create_opts__last_field
+struct bpf_map_create_opts {
```

Comment:
It makes no sense whatsoever to have this stuff downstream. This needs to be provided upstream.

Reply:
This is upstream, from libbpf 0.6.0.
We need something that's called bpf_map_create_opts ad has sz and map_flags fields to please the `LIBBPF_OPTS` call the previous patch added and the function definition just below.

Additionally it's magic and if the struct is close enough, using the libbpf 1.0.0 symbol when compiled with a very old libbpf will also just work so I went all the way with copying it.
Honestly since you're using dlopen it doesn't make much sense not to do that, you're not checking anything.
Either have something that gets proper functions automatically from the headers and I don't care about cross-version compatibility and this can be removed, or admit this system is a mess that doesn't provide any guarantee at which point I don't see the problem with this.

Anyway, I can just say I don't care -- I'll fix the LIBBPF_OPTS macro that I missed to pass checks and won't reply here again.

Reply:
it's the same for all other "compat symbols removed in libbpf 1.0" - the point is that these should have been provided by upstream, it's absurd to force every single user to define them again and again and again

Reply:
i guess they knowingle broke compat there in libbpf1?

This stuff should move to libbpf-compat.h or so, and properly ifdeffed out